### ISSUE TITLE
IDE0052 False Positive when operator ??= is used

### ISSUE THREAD

**Issue by jmo-gelsight** [dotnet/roslyn#59675]
**Version Used**: 

Visual Studio 2022 version 17.1, .NET 6.0.2.  I have the .NET analyzer setting to use "Latest"

**Steps to Reproduce**:

This line of code gives me IDE0052.  

```
_folderDeletedSubscriptionToken ??= AggSvc.GetEvent<FolderDeletedEvent>().Subscribe(
    OnFolderDeleted, ThreadOption.UIThread, false, IsValidDelete);
```

I am trying to remember a private variable that represents a "subscription token" so that it only executes once.  So I store the returned value and use operator `??=` to make the check.

**Expected Behavior**:

I expect no warning.   Operator `??=` _does_ in fact read the value every time it executes

**Actual Behavior**:

I get this warning in the Error List

`Message IDE0052 Private member 'CaptureVm._folderDeletedSubscriptionToken' can be removed as the value assigned to it is never read Modules.Capture `

If I change the line to use an old-style _if value == null then assign_, the warning goes away:
 
```
if (null == _folderDeletedSubscriptionToken)
{
    _folderDeletedSubscriptionToken = AggSvc.GetEvent<FolderDeletedEvent>().Subscribe(
        OnFolderDeleted, ThreadOption.UIThread, false, IsValidDelete);
} 
```

**Comment by sailro** [comment]
Adding another repro to this issue:

```csharp
using System.Collections;

class Repro
{

    private object started;

    void Test()
    {
        started ??= StartRoutine(Routine()); // will trigger IDE0052 on `started`
        //started ??= StartRoutine(nameof(Routine)); // will not trigger IDE0052 on `started`
    }

    object StartRoutine(IEnumerator routine)
    {
        // process stuff given routine

        return "token1";
    }

    object StartRoutine(string routineName)
    {
        // process stuff given routine

        return "token2";
    }

    IEnumerator Routine()
    {
        yield return null;
        started = null;
    }
}
```

**Comment by ryzngard** [comment]
In the initial code posted, this looks to be behaving as expected. IDE0052 not being reported on 

```c#
if (null == _token)
{
	_token = something;
} 
```

is correct because the field is read for comparison. I could see improvements here to report in that case as well...

The second example definitely looks like a bug though. Both of those lines should have the same impact on whether `started` is considered "read".

**Comment by sharwell** [comment]
Duplicate of #66975
